### PR TITLE
[RW-4614][risk=no]integrate jest-puppeteer in Puppeteer e2e test framework

### PR DESCRIPTION
**Pull request by aweng98**

### CODE REVIEW

File: e2e/app/google-login.ts [hunk 1]
```diff
@@ -23,41 +23,51 @@ export default class GoogleLoginPage extends BasePage {
...
+    // compare to the Use another account link
+    const [link] = await this.page.$x(useAnotherAccountXpath);
+    const isLink = await page.evaluate((e1, e2) => e1 === e2, elemt1, link);
+    if (isLink) {
+      // click " Use another Account " link
+      await Promise.all([
+        this.page.waitForNavigation(),
+        link.click(),
+      ]);
     }
```
Comment (aweng98) [inline e2e/app/google-login.ts:64]: Unrelated to jest-puppeteer. This change owns thanks to the unpredictable Google authentication. Test won't fail on Google login page.

File: e2e/tests/user/login.spec.ts [hunk 2]
```diff
@@ -1,40 +1,25 @@
...
-  beforeAll(async () =>  {
-    browser = await PuppeteerLaunch();
-  });
 
   beforeEach(async () => {
-    incognitoContext = await browser.createIncognitoBrowserContext();
-    page = await incognitoContext.newPage();
     await page.setUserAgent(configs.puppeteerUserAgent);
-    await page.setDefaultNavigationTimeout(60000);
+    await page.setDefaultNavigationTimeout(120000);
```
Comment (aweng98) [inline e2e/tests/user/login.spec.ts:10]: For now, `setUserAgent` and `setDefaultNavigationTimeout` are required in `beforeEach`, still is boilerplate code. I have a PR in work that will remove them from test files for good. I didn't want to pack any more changes in this PR.

File: e2e/tests/user/registration-ui.spec.ts [hunk 3]
```diff
@@ -3,56 +3,19 @@ import SelectComponent from '../../app/aou-elements/select-component';
...
   afterEach(async () => {
-    await page.close();
-  });
-
-  afterAll(async () => {
-    await browser.close();
-  });
-
-
-  test('Entered invalid invitation key', async () => {
```
Comment (aweng98) [inline e2e/tests/user/registration-ui.spec.ts:36]: test is no longer valid due to removal of invitation key during Create Account workflow.

File: e2e/tslint.json [hunk 4]
```diff
@@ -1,44 +1,55 @@
 {
-    "defaultSeverity": "error",
-    "extends": [
-        "tslint:recommended",
-        "tslint-eslint-rules",
-        "tslint-config-prettier"
+  "defaultSeverity": "error",
```
Comment (aweng98) [inline e2e/tslint.json:2]: changes in `tslint` are mostly changed by linting. real change are lines #50 - #52.

File: e2e/app/authenticated-page.ts [hunk 5]
```diff
@@ -2,6 +2,7 @@ import {Page} from 'puppeteer';
 import {clrIconXpath} from './aou-elements/xpath-defaults';
 import {findIcon} from './aou-elements/xpath-finder';
 import BasePage from './base-page';
+import {performance} from 'perf_hooks';
```
Comment (calbach) [inline e2e/app/authenticated-page.ts:5]: unused?

File: e2e/app/base-page.ts [hunk 6]
```diff
@@ -232,7 +232,34 @@ export default abstract class BasePage {
     const timestamp = new Date().getTime();
     const screenshotFile = `${SCREENSHOT_DIR}/${fileName}_${timestamp}.png`;
     await this.page.screenshot({path: screenshotFile, fullPage: true});
-    console.log('screenshot taken: ' + screenshotFile);
+    console.log('Saved screenshot ' + screenshotFile);
+  }
+
+  async saveToFile(fileName, data, suffix: string = 'html') {
+    const LOG_DIR = 'logs/html';
```
Comment (calbach) [inline e2e/app/base-page.ts:239]: In the codebase we typically don't capitalize local variables, even if they are constants.

File: e2e/jest-puppeteer.config.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+const lodash = require('lodash');
+const puppeteer = require('puppeteer');
+const isHeadless = process.env.HEADLESS !== 'false';
+
+const NEW_CHROME_SWITCHES = [
```
Comment (calbach) [inline e2e/jest-puppeteer.config.js:5]: How did you decide on these flags - were these copied from elsewhere? Please add a comment.

File: e2e/resources/workbench-config.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 require('dotenv').config();
-const _ = require('lodash');
+const lodash = require('lodash');
```
Comment (calbach) [inline e2e/resources/workbench-config.ts:2]: Existing issue: the rest of the codebase uses lodash/fp, our convention is to alias this as "fp". May want to reconcile this later

File: e2e/tests/app.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import {Page} from 'puppeteer';
+import GoogleLoginPage from '../app/google-login';
+
+
+export const signIn = async (page: Page) => {
+  await GoogleLoginPage.logIn(page);
+   // this element exists in DOM after user has logged in
+  await page.waitFor(() => document.querySelector('app-signed-in app-nav-bar clr-icon') !== null);
```
Comment (calbach) [inline e2e/tests/app.ts:8]: do we need to depend on clr-icon? This seems maybe overly specific. Can we just wait for app-signed-in, for example?

File: e2e/tests/user/registration.spec.ts [hunk 10]
```diff
@@ -44,10 +25,6 @@ describe.skip('User registration tests:', () => {
 
     const createAccountPage = new CreateAccountPage(page);
 
-    // Step 1: Enter invitation key.
-    await createAccountPage.waitForTextExists('Enter your Invitation Key:');
-    await createAccountPage.fillOutInvitationKey(process.env.INVITATION_KEY);
-
     // Step 2: Terms of Service.
```
Comment (calbach) [inline e2e/tests/user/registration.spec.ts:28]: Fix steps

File: e2e/tests/user/registration.spec.ts [hunk 11]
```diff
@@ -1,39 +1,20 @@
...
-  });
-
-  afterAll(async () => {
-    await browser.close();
+    await jestPuppeteer.resetBrowser();
   });
 
 
-  test('Can register new user', async () => {
+  test.skip('Can register new user', async () => {
```
Comment (calbach) [inline e2e/tests/user/registration.spec.ts:17]: Did you mean to leave this in?

File: e2e/tslint.json (hunk 4 above)
Comment (calbach) [reply e2e/tslint.json:2]: Is this the same as our other tslint file? We should probably just have one.

File: e2e/jest-puppeteer.config.js [hunk 12]
```diff
@@ -1,41 +1,45 @@
...
+  [
+    '--disable-gpu', // https://bugs.chromium.org/p/chromium/issues/detail?id=737678#c10
+    '--disable-setuid-sandbox',
+  ]
 );
 
-// switches for Chrome launch
 const SWITCHES = (process.env.CI === 'true') ? CI_SWITCHES : DEFAULT_SWITCHES;
 
+console.log(SWITCHES);
```
Comment (calbach) [inline e2e/jest-puppeteer.config.js:35]: remove?

**Pull request merged**